File: backend/app/schemas/job.py
from pydantic import BaseModel


class JobResponse(BaseModel):
    id: int
    title: str
    company: str
    location: list[str] | None
    remote: bool
    description: str | None
    url: str | None
    field: str | None
    company_industry: list[str] | None
    min_experience: int | None
    requirements: str | None
    required_skills: list[str]
    language_requirement: list[str] | None
    education_requirements: str | None
    posted: str | None


class JobAnalysisRequest(BaseModel):
    job_id: int
    requirements: str


class JobAnalysisItem(BaseModel):
    job_id: int
    required_skills: list[str]


class JobAnalysisBatch(BaseModel):
    jobs: list[JobAnalysisItem]

File: frontend/src/types/job.ts
export interface Job {
  id: number;
  title: string;
  company: string;
  location: string[] | null;
  remote: boolean;
  description: string | null;
  url: string | null;
  field: string | null;
  company_industry: string[] | null;
  min_experience: number | null;
  requirements: string | null;
  required_skills: string[];
  language_requirement: string[] | null;
  education_requirements: string | null;
  posted: string | null;
}

export interface JobsResponse {
  items: Job[];
  total: number;
}

export interface JobCardProps {
  job: Job;
  onClick: () => void;
  match?: {
    score: number;
    matched_skills: string[];
    missing_skills: string[];
  };
}

export interface JobDetailsProps {
  jobId: number;
}
